compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -1035,12 +1039,70 @@ public void visit(BLangUserDefinedType userDefinedTypeNode) {
 
     @Override
     public void visit(BLangFunctionTypeNode functionTypeNode) {
-        List<BType> paramTypes = new ArrayList<>();
-        functionTypeNode.getParams().forEach(t -> paramTypes.add(resolveTypeNode((BLangType) t.getTypeNode(), env)));
-        BType retParamType = resolveTypeNode(functionTypeNode.returnTypeNode, this.env);
-        resultType = new BInvokableType(paramTypes, retParamType, null);
+        resultType = createInvokableType(functionTypeNode.getParams(), functionTypeNode.restParam,
```

Comment:
What if we pass functionTypeNode directly rather than passing individual fields?